### PR TITLE
fix: `no-duplicate-selectors` false positives for SCSS/Less nested interpolations

### PR DESCRIPTION
**Pull request by JounQin**
> Which issue, if any, is this issue related to?

Closes [#6110#issuecomment-1137211409](https://github.com/stylelint/stylelint/issues/6110#issuecomment-1137211409)

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

**Comment by JounQin** [comment]
@ybiquitous Ready for review now.

File: lib/rules/no-duplicate-selectors/index.js [hunk 1]
```diff
@@ -63,19 +63,13 @@ const rule = (primary, secondaryOptions) => {
...
 				),
 			];
 
-			if (resolvedSelectorList.length === 0) {
-				return;
-			}
-
-			const normalizedSelectorList = resolvedSelectorList.map(normalize);
+			const normalizedSelectorList = resolvedSelectorList.map((selector) => {
+				return isStandardSyntaxSelector(selector) ? normalize(selector) : selector;
```
Comment (JounQin) [inline lib/rules/no-duplicate-selectors/index.js:71]: We should not filter them out, because that would change the combined selector semantic.

**Review by ybiquitous — COMMENTED** [review]
@JounQin Thank you! This fix looks good to me 👍🏼 

Also, what about refactoring `normalize()` like this?

```js
function normalize(selector) {
	if (!isStandardSyntaxSelector(selector) return selector;

	return selectorParser().processSync(selector, { lossless: false });
}
```

**Comment by JounQin** [comment]
@ybiquitous Done.

**Review by ybiquitous — APPROVED** [review]
Thank you. LGTM 👍🏼

**Comment by ybiquitous** [comment]
Changelog entry added:

> Fixed: `no-duplicate-selectors` false positives for SCSS/Less nested interpolations ([#6118](https://github.com/stylelint/stylelint/pull/6118)).

**Pull request merged**